hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/connector/UpdateMapWithMaterializedValuesP.java

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.impl.connector;
+
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.function.FunctionEx;
+import com.hazelcast.internal.serialization.SerializationService;
+import com.hazelcast.jet.core.Inbox;
+import com.hazelcast.jet.core.JetDataSerializerHook;
+import com.hazelcast.jet.core.Outbox;
+import com.hazelcast.jet.core.Processor;
+import com.hazelcast.map.EntryProcessor;
+import com.hazelcast.map.IMap;
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.nio.serialization.IdentifiedDataSerializable;
+
+import javax.annotation.CheckReturnValue;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class UpdateMapWithMaterializedValuesP<T, K, V> extends AsyncHazelcastWriterP {
```

Comment:
too much duplication with UpdateMapP

Reply:
My opinion was (and is) that forcing further reuse/code share would hurt understandability greatly, but on the other hand I can definitely give it one more try.